File: app/src/main/java/com/theandroiddev/mywins/ImportancePopupActivity.java
package com.theandroiddev.mywins;

import android.app.Activity;
import android.content.Intent;
import android.os.Build;
import android.os.Bundle;
import android.support.v4.content.ContextCompat;
import android.support.v7.app.AppCompatActivity;
import android.util.DisplayMetrics;
import android.view.View;
import android.widget.Button;
import android.widget.ImageView;
import android.widget.TextView;

import com.theandroiddev.mywins.utils.DrawableSelector;

import static com.theandroiddev.mywins.utils.Constants.dummyImportanceDefault;

public class ImportancePopupActivity extends AppCompatActivity implements View.OnClickListener {
    private static final String TAG = "ImportancePopupActivity";

    TextView popupImportance;
    Button popupDone;
    ImageView importanceIv1, importanceIv2, importanceIv3, importanceIv4;
    private DisplayMetrics displayMetrics;
    private DrawableSelector drawableSelector;

    public ImportancePopupActivity() {
        this.drawableSelector = new DrawableSelector(this);
    }

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_popup_importance);

        initWindow();
        initViews();
    }

    private void initViews() {
        popupImportance = findViewById(R.id.popup_importance_tv);
        popupDone = findViewById(R.id.popup_add_btn);
        importanceIv1 = findViewById(R.id.popup_importance_iv_1);
        importanceIv2 = findViewById(R.id.popup_importance_iv_2);
        importanceIv3 = findViewById(R.id.popup_importance_iv_3);
        importanceIv4 = findViewById(R.id.popup_importance_iv_4);

        importanceIv1.setOnClickListener(this);
        importanceIv2.setOnClickListener(this);
        importanceIv3.setOnClickListener(this);
        importanceIv4.setOnClickListener(this);
        popupDone.setOnClickListener(this);


        int importance = getIntent().getExtras().getInt("importance", 3);
        popupImportance.setText(String.valueOf(importance));


        drawableSelector.setImportance(dummyImportanceDefault, popupImportance, importanceIv1, importanceIv2, importanceIv3, importanceIv4);

        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
            popupDone.setBackgroundTintList(getResources().getColorStateList(R.color.accent, null));
        } else {
            popupDone.setBackgroundColor(ContextCompat.getColor(this, R.color.accent));
        }


    }

    private void saveImportance() {

        Intent returnIntent = new Intent();
        returnIntent.putExtra("importance", drawableSelector.getImportance(popupImportance.getText().toString()));
        setResult(Activity.RESULT_OK, returnIntent);
        finish();
    }

    private void initWindow() {

        displayMetrics = new DisplayMetrics();
        getWindowManager().getDefaultDisplay().getMetrics(displayMetrics);
        int width = displayMetrics.widthPixels;
        int height = displayMetrics.heightPixels;
        getWindow().setLayout((int) (width * .8), (int) (height * .6));

    }

    @Override
    public void onClick(View view) {

        switch (view.getId()) {
            case R.id.popup_importance_iv_1: {

                drawableSelector.setSmallImportance(popupImportance, importanceIv1, importanceIv2, importanceIv3, importanceIv4);
                break;

            }
            case R.id.popup_importance_iv_2:
                drawableSelector.setMediumImportance(popupImportance, importanceIv1, importanceIv2, importanceIv3, importanceIv4);

                break;
            case R.id.popup_importance_iv_3:
                drawableSelector.setBigImportance(popupImportance, importanceIv1, importanceIv2, importanceIv3, importanceIv4);

                break;
            case R.id.popup_importance_iv_4:
                drawableSelector.setHugeImportance(popupImportance, importanceIv1, importanceIv2, importanceIv3, importanceIv4);
                break;
            case R.id.popup_add_btn:
                saveImportance();
                break;

            default:
                break;
        }

    }
}

File: app/src/main/java/com/theandroiddev/mywins/presentation/importance_popup/ImportancePopupActivity.kt
package com.theandroiddev.mywins.presentation.importance_popup

import android.app.Activity
import android.content.Intent
import android.os.Build
import android.os.Bundle
import android.util.DisplayMetrics
import android.view.View
import androidx.core.content.ContextCompat
import com.theandroiddev.mywins.R
import com.theandroiddev.mywins.core.mvp.MvpDaggerAppCompatActivity
import com.theandroiddev.mywins.utils.Constants.Companion.Importance
import com.theandroiddev.mywins.utils.DrawableSelector
import kotlinx.android.synthetic.main.activity_popup_importance.*

class ImportancePopupActivity : MvpDaggerAppCompatActivity<ImportancePopupView,
        ImportancePopupBundle, ImportancePopupPresenter>(), ImportancePopupView, View.OnClickListener {


    private val drawableSelector: DrawableSelector = DrawableSelector(this)
    private var currentImportanceValue = 0

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_popup_importance)

        initWindow()
        initViews()
        presenter.onAfterCreate()

    }

    private fun initViews() {
        popup_importance_iv_1.setOnClickListener(this)
        popup_importance_iv_2.setOnClickListener(this)
        popup_importance_iv_3.setOnClickListener(this)
        popup_importance_iv_4.setOnClickListener(this)
        popup_add_btn.setOnClickListener(this)

    }

    override fun displayInitialImportance(importance: Importance) {

        popup_importance_tv.text = getString(importance.res)
        currentImportanceValue = importance.value
        drawableSelector.setImportance(importance.value, popup_importance_tv, popup_importance_iv_1, popup_importance_iv_2, popup_importance_iv_3, popup_importance_iv_4)

        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
            popup_add_btn.backgroundTintList = resources.getColorStateList(R.color.accent, null)
        } else {
            popup_add_btn.setBackgroundColor(ContextCompat.getColor(this, R.color.accent))
        }

    }

    private fun saveImportance() {

        val returnIntent = Intent()
        returnIntent.putExtra("bundle", ImportancePopupBundle(Importance.values()[currentImportanceValue]))
        setResult(Activity.RESULT_OK, returnIntent)
        finish()
    }

    private fun initWindow() {

        val displayMetrics = DisplayMetrics()
        windowManager.defaultDisplay.getMetrics(displayMetrics)
        val width = displayMetrics.widthPixels
        val height = displayMetrics.heightPixels
        window.setLayout((width * .8).toInt(), (height * .6).toInt())

    }

    override fun onClick(view: View) {

        when (view.id) {
            R.id.popup_importance_iv_1 -> {
                drawableSelector.setSmallImportance(
                        popup_importance_tv,
                        popup_importance_iv_1,
                        popup_importance_iv_2,
                        popup_importance_iv_3,
                        popup_importance_iv_4)
                currentImportanceValue = 1
            }
            R.id.popup_importance_iv_2 -> {
                drawableSelector.setMediumImportance(
                        popup_importance_tv,
                        popup_importance_iv_1,
                        popup_importance_iv_2,
                        popup_importance_iv_3,
                        popup_importance_iv_4)
                currentImportanceValue = 2
            }
            R.id.popup_importance_iv_3 -> {
                drawableSelector.setBigImportance(
                        popup_importance_tv,
                        popup_importance_iv_1,
                        popup_importance_iv_2,
                        popup_importance_iv_3,
                        popup_importance_iv_4)
                currentImportanceValue = 3
            }
            R.id.popup_importance_iv_4 -> {
                drawableSelector.setHugeImportance(
                        popup_importance_tv,
                        popup_importance_iv_1,
                        popup_importance_iv_2,
                        popup_importance_iv_3,
                        popup_importance_iv_4)
                currentImportanceValue = 4
            }
            R.id.popup_add_btn -> {
                saveImportance()
            }

            else -> {
            }
        }

    }
}
